flyway.go

```diff
@@ -0,0 +1,147 @@
+package flyway
+
+import (
+	"context"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"strconv"
+	"time"
+)
+
+const (
+	DefaultFlywayVersion        = "10.10.0"
+	defaultFlywayImagePattern   = "flyway/flyway:%s"
+	defaultFlywayContainerName  = "test_flyway_container"
+	defaultNetworkContainerName = "test_network_container"
+	defaultFlywayUser           = "test_user"
+	defaultFlywayPassword       = "test_password"
+	defaultFlywayDbUrl          = "test_flyway"
+	defaultFlywayTable          = "schema_version"
+	defaultFlywayMigrationsPath = "/flyway/sql"
+	migrateCmd                  = "migrate"
+	infoCmd                     = "info"
+	flywayEnvUserKey            = "FLYWAY_USER"
+	flywayEnvPasswordKey        = "FLYWAY_PASSWORD"
+	flywayEnvUrlKey             = "FLYWAY_URL"
+	flywayEnvGrouopKey          = "FLYWAY_GROUP"
+	flywayEnvTableKey           = "FLYWAY_TABLE"
+	flywayEnvConnectRetriesKey  = "FLYWAY_CONNECT_RETRIES"
+	flywayEnvLocationsKey       = "FLYWAY_LOCATIONS"
+)
+
+// FlywayContainer represents the Flyway container type used in the module
+type FlywayContainer struct {
+	testcontainers.Container
+}
+
+// RunContainer creates an instance of the Flyway container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*FlywayContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Name:       defaultFlywayContainerName,
```

Comment:
We advise against using fixed container names, as it will cause conflicts when running tests in parallel.
```suggestion
```

Reply:
interesting ... I assume it generates a random container name in this case

Reply:
Indeed, the container name is random, and it produces non-deterministic container names, which avoids flakiness caused by resource collision, in this case containers with the same name. That's why testcontainers libraries promotes random free ports too